spec/controllers/hyrax/stats_controller_spec.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe Hyrax::StatsController, type: :controller do
+  routes { HykuAddons::Engine.routes }
+
+  let(:user) { create(:admin) }
+  let(:account) { create(:account) }
+  let(:site) { Site.new(account: account) }
+  let(:gds_fixture) { File.read(HykuAddons::Engine.root.join("spec", "fixtures", "csv", "gds_chart_fixture.csv")) }
+  let(:enabled) { true }
+
+  before do
+    allow(Site).to receive(:instance).and_return(site)
+  end
+
+  describe "GET #reports" do
+    render_views
+
+    before do
+      allow(Flipflop).to receive(:enabled?).and_call_original
+      allow(Flipflop).to receive(:enabled?).with(:gds_reports).and_return(enabled)
+      allow(controller.current_ability).to receive(:can?).with(any_args).and_return(true)
+      allow(account).to receive(:gds_reports).and_return(gds_fixture)
+
+      sign_in(user)
+    end
+
+    context "when the user is an admin" do
+      it "renders a partial" do
+        get :reports
+
+        expect(response).to have_http_status(:ok)
+        expect(response).to render_template(:reports)
+      end
+
+      context "when there is data" do
+        before do
+          get :reports
+        end
+
+        it "displays the content" do
+          expect(response.body).to include(I18n.t("stats.reports.title"))
+          expect(response.body).to include(I18n.t("stats.reports.intro"))
+        end
+
+        it "displays the select" do
+          expect(response.body).to have_selector("#dashaboard_gds_reports")
```

Comment:
I am not sure testing selectors in controller test is the right place

Reply:
I dont love it either, which is why I was looking at system specs, but to add a whole new system spec for 2 expectations, which would take ages to get the permissions setup correctly for, seems like it wouldn't be worth it.

Reply:
Thanks